lib/bolt/inventory/target.rb

```diff
@@ -153,9 +154,24 @@ def validate
           raise Bolt::UnknownTransportError.new(transport, uri)
         end
 
+        validate_fact_names(@facts)
```

Comment:
This needs to be `facts` not `@facts` so that facts set at the group level are also caught, not just those set directly on the target.